montecarlo/eg/src/TDatabasePDG.cxx

```diff
@@ -100,6 +100,33 @@ TDatabasePDG::~TDatabasePDG()
    *fgInstance = nullptr;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Initialize the factory. This makes the GetParticle methods thread safe.
+///
+/// This method forces the initialization of the internal particle database. This
+/// operation has also the effect of avoiding the lazy initialization in the
+/// GetParticle methods, therewith making them safely callable from multiple
+/// threads.
+/// ### Example usage of TDatabasePDG in a thread safe manner:
+///~~~{.cpp}
+///   TDatabasePDG::Init();
+///   auto getParticle = [](const char *pname) {
+///   auto particle = TDatabasePDG::Instance()->GetParticle(pname);
+///   // Do something with particle
+///   };
+///   ROOT::TThreadExecutor pool(4);
+///   mypool.Foreach(getParticle, {"g", "gamma", "Z0", "W+", "h0"}});
+///~~~
+
+void TDatabasePDG::Init()
+{
```

Comment:
Would it make sense to move this initialization into the static `Instance` method which guarantees that the initialization happens exactly once, during the initialization of the static local instance variable? Then any other method such as `TDatabasePDG::GetParticle()` would automatically be thread safe, without any changes to user code.